packages/docs-markdown/src/extension.ts

```diff
@@ -201,10 +201,8 @@ export async function activate(context: ExtensionContext) {
 			}
 		}
 	});
-	window.onDidChangeActiveTextEditor(e => {
-		if (['markdown', 'json'].includes(e.document.languageId)) {
-			metadataTreeProvider.refresh();
-		}
+	workspace.onDidCloseTextDocument(e => {
+		metadataTreeProvider.refresh();
```

Comment:
When the explorer refreshes, it checks the language, so we don't need to check here. Also, we want to update the explorer even if the active document isn't Markdown (i.e. show the welcome message).